curation_concerns-models/app/jobs/ingest_file_job.rb

```diff
@@ -3,13 +3,27 @@ class IngestFileJob < ActiveJob::Base
 
   def perform(file_set_id, filename, mime_type, user_key)
     file_set = FileSet.find(file_set_id)
-    file = Hydra::Derivatives::IoDecorator.new(File.open(filename, "rb"))
-    file.mime_type = mime_type
-    file.original_name = File.basename(filename)
 
-    # Tell UploadFileToGenericFile service to skip versioning because versions will be minted by VersionCommitter (called by save_characterize_and_record_committer) when necessary
-    Hydra::Works::UploadFileToFileSet.call(file_set, file, versioning: false)
+    file = File.open(filename, "rb")
+    # If mime-type is known, wrap in an IO decorator
+    # Otherwise allow Hydra::Works service to determine mime_type
+    if mime_type
+      file = Hydra::Derivatives::IoDecorator.new(file)
+      file.mime_type = mime_type
+      file.original_name = File.basename(filename)
+    end
+
+    # Tell AddFileToFileSet service to skip versioning because versions will be minted by VersionCommitter (called by save_characterize_and_record_committer) when necessary
+    Hydra::Works::AddFileToFileSet.call(file_set, file, :original_file, versioning: false)
+
+    # Assign label and title of File Set after original_name has been determined.
+    file_set.label ||= file_set.original_file.original_name
+    file_set.title = [file_set.label] if file_set.title.blank?
```

Comment:
Setting file_set.title here does not get persisted according to the specs, and I'm baffled as to why that is the case.